app/src/main/java/com/example/countryinfo/CountryDetailsViewModel.kt

```diff
@@ -34,10 +36,113 @@ class CountryDetailsViewModel(private val countryApi: ICountriesApi) : ViewModel
             }
 
         disposable = countryObservable.subscribe(
-            { country -> countryMutableLiveData.postValue(CountryDetailsViewState.Default(country)) },
+            { country ->
+                countryMutableLiveData.postValue(
+                    CountryDetailsViewState.Default(
+                        country,
+                        setCountryStates(country)
+                    )
+                )
+            },
             { error -> countryMutableLiveData.postValue(CountryDetailsViewState.Error(error)) })
     }
 
+    private fun setCountryStates(countryDetails: CountryDetails): List<DetailsViewHolderState> {
+        val state: MutableList<DetailsViewHolderState> = mutableListOf(DetailsViewHolderState())
```

Comment:
[nit] You can initialise an immutable list instead:
```suggestion
        val state = listOf(
            DetailsViewHolderState(
                 DetailsSections.COUNTRY_NAME.title,
                 listOf(countryDetails.name),
                 false
             ),
             DetailsViewHolderState(
                 DetailsSections.CAPITAL.title,
                 listOf(countryDetails.capital),
                 false
             ),
             ....
        )
```

Reply:
Done.